azure-pipelines.yml

```diff
@@ -0,0 +1,156 @@
+# azure-pipelines build file
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema
+
+# TODO: Add gh-pages deployment on tag trigger
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - master
+    - RB-*
+    - ci_test
+
+pr:
+  autoCancel: true
+  branches:
+    include:
+    - master
+    - RB-*
+
+# Common variables
+variables:
+  cxxStd: 14
+  nixCxxFlags: -std=c++$(cxxStd) -fPIC
+
+jobs:
+
+# ------------------------------------------------------------------------------
+# Linux
+# ------------------------------------------------------------------------------
+# TODO: Move Docker images to ASWF DockerHub account
+
+- job: Linux
+  strategy:
+    matrix:
+      Ubuntu 18.04 VFX CY2020:
+        agentImage: 'ubuntu-16.04'
+        containerImage: michdolan/ubuntu18.04:vfx2020_py27
+        cxxFlags: ''
+        cxxCompiler: g++
+        cCompiler: gcc
+      Ubuntu 18.04 VFX CY2019:
+        agentImage: 'ubuntu-16.04'
+        containerImage: michdolan/ubuntu18.04:vfx2019
+        cxxFlags: ''
+        cxxCompiler: g++
+        cCompiler: gcc
+      Ubuntu 18.04 Clang 6.0:
+        agentImage: 'ubuntu-16.04'
+        containerImage: michdolan/ubuntu18.04:vfx2019
+        cxxFlags: -Wno-deprecated-register
+        cxxCompiler: clang++
+        cCompiler: clang
+  pool:
+    vmImage: $(agentImage)
+  container: $[ variables['containerImage'] ]
+
+  steps:
+  - template: share/ci/templates/checkout.yml
+  - template: share/ci/templates/configure.yml
+    parameters:
+      cxxFlags: $(nixCxxFlags) $(cxxFlags)
+      cxxCompiler: $(cxxCompiler)
+      cCompiler: $(cCompiler)
+
+  - template: share/ci/templates/build.yml
+    parameters:
+      cxxCompiler: $(cxxCompiler)
+      cCompiler: $(cCompiler)
+      cmakeOpts: -- -j2
+
+# --------------------------
+# SonarCloud static analysis
+# --------------------------
+- job: SonarCloud
+  displayName: SonarCloud Ubuntu 18.04 VFX CY2019
+  condition: in(variables['Build.Reason'], 'Manual', 'Schedule')
+  pool:
+    vmImage: 'ubuntu-16.04'
+  container: michdolan/ubuntu18.04:vfx2019
+
+  steps:
+  - template: share/ci/templates/checkout.yml
+  - template: share/ci/templates/configure.yml
+    parameters:
+      cxxFlags: $(nixCxxFlags) -g -O0 -fprofile-arcs -ftest-coverage
```

Comment:
This was a point I meant to raise for discussion. I set the c++ std to 14 globally (though our CMake setup forces 11 for OCIO) because VFX Reference Platform 18/19/20 all specify 14. What are your thoughts on allowing OCIO to be built with a user defined standard, just defaulting to 11? OIIO does that - defaults to 11, but supports 14 or 17 being specified.

Reply:
Agree